gcp/api/server.py

```diff
@@ -197,6 +197,24 @@ def query_by_commit(commit, to_response=bug_to_response):
   return _get_bugs(bug_ids, to_response=to_response)
 
 
+def _match_purl(purl_query, purl_db) -> bool:
```

Comment:
I'm not entirely clear on purls and what you're trying to accomplish here, but I was wondering if utilising the likes of https://urllib3.readthedocs.io/en/stable/reference/urllib3.util.html#urllib3.util.parse_url might be a more robust option?

Reply:
That's a good point about making the parsing more robust, but we should use the offiicial package URL module here if we can (which we already import in this file in https://github.com/google/osv.dev/blob/3f8b4067bb83dfe9e872a8b7bae584f9e8821f12/gcp/api/server.py#L26)

Reply:
[f1b4443](https://github.com/google/osv.dev/pull/589/commits/f1b4443d240bfff5f1463ba03fbbd5d0b4972bba) Now fixes this by using PackageURL for the purl variable being passed around.